File: C++ Programming/OOPX.cpp
////////////////////////////////////////////////////////////////////////////////////////////////
//
//  File name :     OOPX.cpp
//  Description :   Demonstrates constructors in C++ by using both default
//                  and parameterized constructors within a class to perform
//                  arithmetic operations.
//  Author :        Varad Nitin Muley
//  Date :          17/10/2025
//
////////////////////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////
//
//  Required Header files
//
/////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

/////////////////////////////////////////////////////////////////
//
//  Class Name : Arithematic
//  Description : Performs addition and subtraction using
//                class and constructor concepts.
//
/////////////////////////////////////////////////////////////////

class Arithematic
{
    public:
        int iNo1;
        int iNo2;

        Arithematic()                         // Default constructor
        {
            cout<<"Inside default constructor \n";
            iNo1 = 0;
            iNo2 = 0;
        }

        Arithematic(int A, int B)             // Parameterized constructor
        {
            cout<<"Inside Parameterized constructor \n";
            iNo1 = A;
            iNo2 = B;
        }

        int Addition()
        {
            int iAns = 0;
            iAns = iNo1 + iNo2;               // Addition logic
            return iAns;
        }

        int Substraction()
        {
            int iAns = 0;
            iAns = iNo1 - iNo2;               // Subtraction logic
            return iAns;
        }
};

/////////////////////////////////////////////////////////////////
//
//  Entry point function for the application
//
/////////////////////////////////////////////////////////////////

int main()
{
    Arithematic aobj1(11, 10);                // Parameterized object
    Arithematic aobj2;                        // Default object

    int iRet = 0;

    iRet = aobj1.Addition();                  // Function call
    cout<<"Addition is : "<<iRet<<"\n";

    iRet = aobj1.Substraction();              // Function call
    cout<<"Subtraction is : "<<iRet<<"\n";

    return 0;
}   // End of main

/////////////////////////////////////////////////////////////////
//
//  Testcase successfully handled by the application
//
//  Output : Inside Parameterized constructor
//           Inside default constructor
//           Addition is : 21
//           Subtraction is : 1
//
/////////////////////////////////////////////////////////////////

File: Java Programming/OOPX.java
////////////////////////////////////////////////////////////////////////////////////////////////
//
//  File name :     OOPX.java
//  Description :   Demonstrates object-oriented programming concepts
//                  using a class that performs arithmetic operations.
//  Author :        Varad Nitin Muley
//  Date :          24/11/2025
//
////////////////////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////
//
//  Class Name :    Arithematic
//  Description :   Performs basic arithmetic operations such as
//                  addition and subtraction using instance data.
//
/////////////////////////////////////////////////////////////////

class Arithematic
{
    public int iNo1; 
    public int iNo2;

    /////////////////////////////////////////////////////////////////
    //
    //  Constructor Name : Arithematic
    //  Description      : Default constructor initializes values to zero.
    //
    /////////////////////////////////////////////////////////////////

    public Arithematic()
    {
        this.iNo1 = 0;
        this.iNo2 = 0;
    }

    /////////////////////////////////////////////////////////////////
    //
    //  Constructor Name : Arithematic
    //  Description      : Parameterized constructor initializes values.
    //  Input            : Integer, Integer
    //
    /////////////////////////////////////////////////////////////////

    public Arithematic(int i, int j)
    {
        this.iNo1 = i;
        this.iNo2 = j;
    }

    /////////////////////////////////////////////////////////////////
    //
    //  Function Name : Addition
    //  Description   : Returns addition of two numbers.
    //  Input         : void
    //  Output        : Integer
    //
    /////////////////////////////////////////////////////////////////

    public int Addition()
    {
        int iAns = 0;
        iAns = this.iNo1 + this.iNo2;
        return iAns;
    }

    /////////////////////////////////////////////////////////////////
    //
    //  Function Name : Substraction
    //  Description   : Returns subtraction of two numbers.
    //  Input         : void
    //  Output        : Integer
    //
    /////////////////////////////////////////////////////////////////

    public int Substraction()
    {
        int iAns = 0;
        iAns = this.iNo1 - this.iNo2;
        return iAns;
    }
}

/////////////////////////////////////////////////////////////////
//
//  Class Name :    OOPX
//  Description :   Entry point class to demonstrate usage of
//                  Arithematic class.
//
/////////////////////////////////////////////////////////////////

class OOPX
{
    /////////////////////////////////////////////////////////////////
    //
    //  Function Name : main
    //  Description   : Entry point of Java application
    //  Input         : String array
    //  Output        : void
    //
    /////////////////////////////////////////////////////////////////

    public static void main(String Arr[])
    {
        Arithematic aobj1 = new Arithematic(11, 10);

        int iRet = 0;

        iRet = aobj1.Addition();
        System.out.println("Addition is : " + iRet); 

        iRet = aobj1.Substraction();
        System.out.println("Substraction is : " + iRet); 

        aobj1 = null;
    }
}

/////////////////////////////////////////////////////////////////
//
//  Testcases:
//
//  Object Creation : Arithematic aobj1 = new Arithematic(11,10);
//
//  Function Call   : Addition()
//  Output          : Addition is : 21
//
//  Function Call   : Substraction()
//  Output          : Substraction is : 1
//
/////////////////////////////////////////////////////////////////
